src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -688,20 +689,26 @@ private void WaitForJournalStateToBeUpdated(CancellationToken token, Transaction
                         {
                             break;
                         }
+                        catch (EarlyOutOfMemoryException)
+                        {
+                            // we already finished the flush process, the next step is to update
+                            // the journal state and free up the scratch buffer pool used pages.
+                            // after this process is done we'll be able to delete the scratch buffers
+                            // that we flushed and start the sync process
+
+                            if (CanContinueWaiting(token))
```

Comment:
I'm not following how this is helpful for us.
We attempted to allocated some native memory (why, I can't think of a code path here that needs it) and failed.
why are we waiting instead of just failing upward?
Is this because it is potentially expected?

Reply:
We try to allocate memory when opening a write tx.
See the stacktrace here: http://issues.hibernatingrhinos.com/issue/RavenDB-12286
In this case we aren't going to do any work using this write transaction (we hold it to prevent others to open a write tx). Can we disable this low memory check here?
Failing here will result in VoronUnrecoverableErrorException to the storage environment.

Reply:
If we don't need the write tx, what about simply providing API to get the write tx lock, instead?
I like it better than starting to mess with this here, and IIRC, there are other places where we start a write tx just to be able to get the lock

Reply:
done